### ISSUE TITLE
environment variable evaluated as template --> fatal error

### ISSUE THREAD

**Issue by holmberg556** [ansible/ansible#12469]
##### Summary:

Ansible seems to require that all environment variables can be evaluated as templates. I have a simple case where the value of a "random" environment variable (SOME_NAME) in my shell causes a fatal error in ansible:

```
fatal: [localhost] => Failed to template %(t.Ding!.%D{%L:%M})%# : template error while templating string: Encountered unknown tag 'L'.
```

If I unset SOME_NAME in the shell (and don't use a fancy Zsh prompt) the playbook runs fine.
##### Steps To Reproduce:

The following setting in my shell causes a the fatal error in ansible (the value was taken from part of my normal zsh prompt):

```
export SOME_NAME='%(t.Ding!.%D{%L:%M})%# '
```

A minimal playbook I used to trigger the problem was:

```
- name: demonstrate ansible bug
  hosts: localhost

  tasks:
    - name: copy a file
      copy: src=/etc/hosts
            dest=/tmp/etc-hosts-copy
```
##### Expected Results:

The playbook should run without any error.
##### Actual Results:

The error look like:

```
TASK: [copy a file] *********************************************************** 
fatal: [localhost] => Failed to template %(t.Ding!.%D{%L:%M})%# : template error while templating string: Encountered unknown tag 'L'.

FATAL: all hosts have already failed -- aborting
```
##### Comments:

I have seen the Issue #6464 that describes almost the same problem (it seems to me). But it was closed, so therefore I wrote this new issue.

**Comment by abadger** [comment]
Note:  This is fixed in ansible-2.0.  adding to the stable-1.9 milestone.

**Comment by abadger** [comment]
Just tested this and it's been fixed with this commit: edf3164bc79d8813c9919ca657d52228618e0dd4 .  Will be in the 1.9.5 release when we release it (should be soon).
